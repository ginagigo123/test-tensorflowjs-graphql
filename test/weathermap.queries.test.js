import { createTestClient } from 'apollo-server-testing'
import { gql } from 'apollo-server'
import graphqlServer from '../graphql'

const GET_COUNTRY_WEATHER = gql`
query Weather($country: String!) {
  weather(country: $country) {
    message
    cod
    count
    list {
      id
      name
      dt
      rain
      snow
      weather {
        id
        main
        description
        icon
      }
      clouds {
        all
      }
      sys {
        country
      }
      wind {
        speed
        deg
      }
      main {
        temp
        feels_like
        temp_min
        temp_max
        pressure
        humidity
      }
      coord {
        lat
        lon
      }
    }
  }
}
`

describe('[Queries.WeatherMapAPI]', () => {
  it('Access current weather data for any location', async () => {
    const { query } = createTestClient(graphqlServer)

    const res = await query({
      query: GET_COUNTRY_WEATHER,
      variables: { source: 'London' }
    })

    expect(res).toMatchSnapshot()
  })
})
