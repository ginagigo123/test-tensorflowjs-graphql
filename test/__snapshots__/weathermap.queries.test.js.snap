// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Queries.WeatherMapAPI] Access current weather data for any location 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [UserInputError: Variable "$country" of required type "String!" was not provided.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
